File: javadevelopers-understanding-kotlin-properties/src/test/java/com/moelholm/jav/SecondTests.java
package com.moelholm.jav;

import static org.assertj.core.api.Assertions.assertThat;

import com.moelholm.Second;
import org.junit.Test;

public class SecondTests {

  @Test
  public void test() {

    Second second = new Second("Hello from Java", "Hello again from Java");

    assertThat(second.getMessageOne()).isEqualTo("Hello from Java");
    assertThat(second.getMessageTwo()).isEqualTo("Hello again from Java");

    second.setMessageTwo("You are mutable :)");

    assertThat(second.getMessageTwo()).isEqualTo("You are mutable :)");

  }

}

File: javadevelopers-understanding-kotlin-properties/src/test/kotlin/com/moelholm/kot/SecondTests.kt
package com.moelholm.kot

import com.moelholm.Second
import org.assertj.core.api.Assertions.assertThat
import org.junit.Test

class SecondTests {

  @Test
  fun test() {

    val second = Second("Hello from Kotlin", "Hello again from Kotlin")

    assertThat(second.messageOne).isEqualTo("Hello from Kotlin")
    assertThat(second.messageTwo).isEqualTo("Hello again from Kotlin")

    second.messageTwo = "You are mutable :)"

    assertThat(second.messageTwo).isEqualTo("You are mutable :)")

  }

}